src/readii/io/loaders/features.py

```diff
@@ -39,28 +42,44 @@ def loadFeatureFilesFromImageTypes(extracted_feature_dir:str,
     feature_sets = {}
 
     # Check if the passed in extracted feature directory exists
-    if not os.path.isdir(extracted_feature_dir):
-        raise FileNotFoundError(f"Extracted feature directory {extracted_feature_dir} does not exist.")
+    if not extracted_feature_dir.exists() or not extracted_feature_dir.is_dir():
+        msg = f"Extracted feature directory {extracted_feature_dir} does not exist."
+        logger.error(f"Extracted feature directory {extracted_feature_dir} does not exist.")
+        raise FileNotFoundError()
     
-    feature_file_list = os.listdir(extracted_feature_dir)
+    feature_file_list = sorted(extracted_feature_dir.glob("*.csv"))
 
     # Loop through all the files in the directory
     for image_type in image_types:
         try:
             # Extract the image type feature csv file from the feature directory  
-            matching_files = [file for file in feature_file_list if (image_type in file) and (file.endswith(".csv"))]  
-            if matching_files:  
-                image_type_feature_file = matching_files[0]  
-                # Remove the image type file from the list of feature files  
-                feature_file_list.remove(image_type_feature_file)
-        except IndexError:
-            logger.warning(f"No {image_type} feature csv files found in {extracted_feature_dir}")
-            # Skip to the next image type
-            continue
+            matching_files = [file for file in feature_file_list if (image_type in file)]  
+
+            match len(matching_files):
+                case 1:
+                    # Only one file found, use it  
+                    pass
+                case 0:
+                    # No files found for this image type
+                    logger.warning(f"No {image_type} feature csv files found in {extracted_feature_dir}")
+                    # Skip to the next image type
+                    continue
+                case _:
+                    # Multiple files found
+                    msg = f"Multiple {image_type} feature csv files found in {extracted_feature_dir}. First one will be used."
+                    logger.warning(msg)
+
+            image_type_feature_file = matching_files[0]  
+            # Remove the image type file from the list of feature files  
+            feature_file_list.remove(image_type_feature_file)
+
+        except Exception as e:
+            logger.warning(f"Error loading {image_type} feature csv files from {extracted_feature_dir}: {e}")
+            raise e
```

Comment:
what kind of exceptions do you forsee in this `try-except` block? Looks like the most possible source of errors are well handled i.e no chance for index error